File: frontend/src/api/psychology.ts
import { request } from './request'

export interface Scale {
  id: number
  code: string
  name: string
  target: string
  description: string | null
  question_count: number
  sort_order: number
}

export interface ScaleQuestion {
  id: number
  sort_order: number
  content: string
  options: { label: string; score: number }[]
  dimension: string | null
  reverse: boolean
}

export interface ScaleDetail {
  id: number
  code: string
  name: string
  target: string
  description: string | null
  interpretation: { min: number; max: number; level: string; color: string; advice: string }[]
  questions: ScaleQuestion[]
}

export interface AssessmentItem {
  id: number
  student_id: number
  student_name: string | null
  scale_id: number
  scale_code: string
  scale_name: string
  total_score: number
  level: string
  level_color: string
  advice: string | null
  dimension_scores: Record<string, number>
  completed_at: string | null
}

export interface TextAnalysisItem {
  id: number
  student_id: number
  title: string | null
  content: string
  polarity: string
  risk_level: string
  risk_keywords: string[]
  emotion_tags: string[]
  summary: string | null
  suggestion: string | null
  created_at: string
}

export interface ConversationItem {
  id: number
  student_id: number
  title: string
  risk_level: string
  message_count: number
  created_at: string
}

export interface MessageItem {
  id: number
  role: 'user' | 'assistant' | 'system'
  content: string
  risk_level: string | null
  risk_keywords: string[]
  created_at: string
}

export interface TimelinePoint {
  date: string
  score: number
  polarity: string
  risk_level: string
  source: string
  note: string | null
}

export interface PsychologyProfile {
  student: { id: number; name: string; class_id: number }
  latest_score: number
  top_risk: string
  assessments: AssessmentItem[]
  text_analyses: TextAnalysisItem[]
  conversations: ConversationItem[]
  timeline: TimelinePoint[]
}

export const listScales = () =>
  request<{ items: Scale[]; total: number }>({ url: '/psychology/scales', method: 'get' })

export const getScale = (id: number) =>
  request<ScaleDetail>({ url: `/psychology/scales/${id}`, method: 'get' })

export const seedScales = () =>
  request<{ seeded: number }>({ url: '/psychology/scales/seed', method: 'post' })

export const submitAssessment = (data: any) =>
  request<AssessmentItem>({ url: '/psychology/assessments', method: 'post', data })

export const listAssessments = (params?: any) =>
  request<{ items: AssessmentItem[]; total: number }>({
    url: '/psychology/assessments',
    method: 'get',
    params,
  })

export const analyzeText = (data: any) =>
  request<TextAnalysisItem>({ url: '/psychology/text-analyses', method: 'post', data })

export const listTextAnalyses = (studentId: number) =>
  request<{ items: TextAnalysisItem[]; total: number }>({
    url: '/psychology/text-analyses',
    method: 'get',
    params: { student_id: studentId },
  })

export const startConversation = (studentId: number, title?: string) =>
  request<ConversationItem>({
    url: '/psychology/conversations',
    method: 'post',
    data: { student_id: studentId, title },
  })

export const listConversations = (studentId?: number) =>
  request<{ items: ConversationItem[]; total: number }>({
    url: '/psychology/conversations',
    method: 'get',
    params: studentId ? { student_id: studentId } : {},
  })

export const getConversation = (id: number) =>
  request<ConversationItem & { messages: MessageItem[] }>({
    url: `/psychology/conversations/${id}`,
    method: 'get',
  })

export const postMessage = (convId: number, content: string) =>
  request<{ user_message: MessageItem; assistant_message: MessageItem; conversation: ConversationItem }>({
    url: `/psychology/conversations/${convId}/messages`,
    method: 'post',
    data: { content },
  })

export const getProfile = (studentId: number) =>
  request<PsychologyProfile>({
    url: `/psychology/students/${studentId}/profile`,
    method: 'get',
  })

export const getTimeline = (studentId: number, days = 60) =>
  request<{ items: TimelinePoint[] }>({
    url: `/psychology/students/${studentId}/timeline`,
    method: 'get',
    params: { days },
  })

File: backend/app/api/psychology.py
"""心理健康接口。"""

from __future__ import annotations

from flask import Blueprint, request

from ..services import psychology_service as svc
from ..utils.permissions import (
    assert_can_access_student,
    get_current_user,
    login_required,
)
from ..utils.response import ok

psy_bp = Blueprint("psychology", __name__)


# ---- 量表 ----
@psy_bp.get("/scales")
@login_required
def list_scales():
    items = svc.list_scales()
    return ok({"items": items, "total": len(items)})


@psy_bp.get("/scales/<int:scale_id>")
@login_required
def get_scale(scale_id: int):
    return ok(svc.get_scale_with_questions(scale_id))


@psy_bp.post("/scales/seed")
@login_required
def seed_scales():
    cnt = svc.ensure_scales_seeded()
    return ok({"seeded": cnt}, "已初始化")


# ---- 测评 ----
@psy_bp.post("/assessments")
@login_required
def submit_assessment():
    user = get_current_user()
    payload = request.get_json() or {}
    student_id = int(payload.get("student_id"))
    assert_can_access_student(user, student_id)
    answers = payload.get("answers") or {}
    return ok(
        svc.submit_assessment(
            student_id=student_id,
            scale_id=int(payload.get("scale_id")),
            answers={int(k): int(v) for k, v in answers.items()},
            operator_id=user.id,
        ),
        "提交成功",
    )


@psy_bp.get("/assessments")
@login_required
def list_assessments():
    user = get_current_user()
    student_id = request.args.get("student_id", type=int)
    scale_id = request.args.get("scale_id", type=int)
    if student_id:
        assert_can_access_student(user, student_id)
    items = svc.list_assessments(student_id=student_id, scale_id=scale_id)
    return ok({"items": items, "total": len(items)})


# ---- 文本分析 ----
@psy_bp.post("/text-analyses")
@login_required
def analyze_text():
    user = get_current_user()
    payload = request.get_json() or {}
    student_id = int(payload.get("student_id"))
    assert_can_access_student(user, student_id)
    return ok(
        svc.analyze_text(
            student_id=student_id,
            content=payload.get("content", ""),
            title=payload.get("title"),
            operator_id=user.id,
        ),
        "已分析",
    )


@psy_bp.get("/text-analyses")
@login_required
def list_text_analyses():
    user = get_current_user()
    student_id = request.args.get("student_id", type=int)
    if not student_id:
        return ok({"items": [], "total": 0})
    assert_can_access_student(user, student_id)
    items = svc.list_text_analyses(student_id)
    return ok({"items": items, "total": len(items)})


# ---- AI 对话 ----
@psy_bp.post("/conversations")
@login_required
def start_conversation():
    user = get_current_user()
    payload = request.get_json() or {}
    student_id = int(payload.get("student_id"))
    assert_can_access_student(user, student_id)
    return ok(svc.start_conversation(student_id, user.id, payload.get("title")), "已创建")


@psy_bp.get("/conversations")
@login_required
def list_conversations():
    user = get_current_user()
    student_id = request.args.get("student_id", type=int)
    if student_id:
        assert_can_access_student(user, student_id)
    return ok({"items": svc.list_conversations(student_id), "total": 0})


@psy_bp.get("/conversations/<int:conv_id>")
@login_required
def get_conversation(conv_id: int):
    return ok(svc.get_conversation_detail(conv_id))


@psy_bp.post("/conversations/<int:conv_id>/messages")
@login_required
def post_message(conv_id: int):
    payload = request.get_json() or {}
    return ok(svc.post_message(conv_id, payload.get("content", "")))


# ---- 个人档案 ----
@psy_bp.get("/students/<int:student_id>/profile")
@login_required
def student_profile(student_id: int):
    user = get_current_user()
    assert_can_access_student(user, student_id)
    return ok(svc.student_psychology_profile(student_id))


@psy_bp.get("/students/<int:student_id>/timeline")
@login_required
def student_timeline(student_id: int):
    user = get_current_user()
    assert_can_access_student(user, student_id)
    days = request.args.get("days", default=60, type=int)
    return ok({"items": svc.student_emotion_timeline(student_id, days=days)})
